### PR TITLE
[Confluent] Init release

### PR DESCRIPTION
**Pull request by fengzhou-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Confluent

File: src/confluent/azext_confluent/azext_metadata.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isExperimental": true,
+    "azext.minCliCoreVersion": "2.11.0"
```
Comment (fengzhou-msft) [inline src/confluent/azext_confluent/azext_metadata.json:3]: Need to change min cli version to `2.18.0` after it's released.

File: src/confluent/azext_confluent/generated/_help.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+helps['confluent term'] = """
+    type: group
+    short-summary: Manage marketplace agreement with confluent
+"""
+
+helps['confluent term list'] = """
+    type: command
+    short-summary: "List Confluent marketplace agreements in the subscription."
+    examples:
+      - name: MarketplaceAgreements_List
```
Comment (qwordy) [inline src/confluent/azext_confluent/generated/_help.py:24]: Please update description in manual folder.

File: src/confluent/report.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+# Azure CLI Module Creation Report
```
Comment (qwordy) [inline src/confluent/report.md:1]: Is this file required?

File: src/confluent/azext_confluent/tests/latest/example_steps.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# --------------------------------------------------------------------------
```
Comment (qwordy) [inline src/confluent/azext_confluent/tests/latest/example_steps.py:1]: Tests are in two files. example_step.py, test_confluent_scenario.py. Is example_step.py just example code and should be deleted?

File: src/confluent/azext_confluent/custom.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# --------------------------------------------------------------------------
```
Comment (qwordy) [inline src/confluent/azext_confluent/custom.py:1]: Should custom.py be in generated or manual folder?

File: src/confluent/report.md (hunk 3 above)
Comment (fengzhou-msft) [reply src/confluent/report.md:1]: Generated and used by codegen, may be helpful for incremental codegen.

File: src/confluent/azext_confluent/tests/latest/example_steps.py (hunk 4 above)
Comment (fengzhou-msft) [reply src/confluent/azext_confluent/tests/latest/example_steps.py:1]: The test is refactored in this way by codegen to better support both generated code and manual code for tests (you can add manual code to replace a step in the tests). `example_step.py` contains the actual operations generated from swagger test examples and is used in `test_confluent_scenario.py`.

File: src/confluent/azext_confluent/custom.py (hunk 5 above)
Comment (fengzhou-msft) [reply src/confluent/azext_confluent/custom.py:1]: I did not modify this file after it's autogenerated, so it's only in the generated folder.

**Review by qwordy — APPROVED** [review]

File: src/confluent/azext_confluent/azext_metadata.json (hunk 1 above)
Comment (fengzhou-msft) [inline src/confluent/azext_confluent/azext_metadata.json:3]:
```suggestion
    "azext.minCliCoreVersion": "2.17.0"
```
